mypy/semanal_typeddict.py

```diff
@@ -316,6 +324,7 @@ def analyze_typeddict_classdef_fields(
                     analyzed = self.api.anal_type(
                         stmt.type,
                         allow_required=True,
+                        allow_unpack=True,
```

Comment:
Leaving `allow_unpack` out here, or setting it to `False`, doesn't have an effect on the test suite... Intuitively I would guess that allow_unpack should be `True` here but would love to understand this a little bit better.